Library/Homebrew/cask/cask_loader.rb

```diff
@@ -214,15 +214,17 @@ def load(config:)
 
         # Use the cask-source API if there are any `*flight` blocks
         if json_cask[:artifacts].any? { |artifact| FLIGHT_STANZAS.include?(artifact.keys.first) }
-          cask_source = Homebrew::API::CaskSource.fetch(token)
+          cask_source = Homebrew::API::CaskSource.fetch(token, git_head: json_cask[:tap_git_head])
           return FromContentLoader.new(cask_source).load(config: config)
         end
 
         # convert generic string replacements into actual ones
         json_cask[:artifacts] = json_cask[:artifacts].map(&method(:from_h_hash_gsubs))
         json_cask[:caveats] = from_h_string_gsubs(json_cask[:caveats])
 
-        Cask.new(token, source: cask_source, config: config) do
+        tap = Tap.fetch(json_cask[:tap]) if json_cask[:tap].to_s.include?("/")
```

Comment:
What's the case where `json_cask[:tap].to_s` does not include `/`?

Reply:
Mainly thinking of the case where it's an empty/invalid string.